dm: update doc for error handling (#7894)
pingcap/docs: dm/handle-failed-ddl-statements.md

This document introduces how to handle failed DDL statements when you're using the TiDB Data Migration (DM) tool to migrate data.

Currently, TiDB is not completely compatible with all MySQL syntax (see [the DDL statements supported by TiDB](/mysql-compatibility.md#ddl)). Therefore, when DM is migrating data from MySQL to TiDB and TiDB does not support the corresponding DDL statement, an error might occur and break the migration process. In this case, you can use the `binlog` command of DM to resume the migration.

## Restrictions

Do not use this command in the following situations:

* It is unacceptable in the actual production environment that the failed DDL statement is skipped in the downstream TiDB.
* The failed DDL statement cannot be replaced with other DDL statements.

> * Other DDL statements must not be injected into the downstream TiDB.

Reviewer: 这段话比较费解，双重否定比较多，读者被绕晕了。建议按照中文重新翻译。

> Do not use this command in the following situations:
> 
> * It is unacceptable in the actual production environment that the failed DDL statement is skipped in the downstream TiDB.
> * The failed DDL statement cannot be replaced with other DDL statements.
> * Other DDL statements must not be injected into the downstream TiDB.

Author: ```suggestion
If it is unacceptable in the actual production environment that the failed DDL statement is skipped in the downstream TiDB, if the DDL statement cannot be replaced with other DDL statements, and if it is unacceptable to inject other DDL statements, do not use this command.
```

Author: @hfxsd PTAL

## Supported scenarios

During the migration, the DDL statement unsupported by TiDB is executed in the upstream and migrated to the downstream, and as a result, the migration task gets interrupted.

- If it is acceptable that this DDL statement is skipped in the downstream TiDB, then you can use `binlog skip <task-name>` to skip migrating this DDL statement and resume the migration.
- If it is acceptable that this DDL statement is replaced with other DDL statements, then you can use `binlog replace <task-name>` to replace this DDL statement and resume the migration.
- If it is acceptable that other DDL statements are injected to the downstream TiDB, then you can use `binlog inject <task-name>` to inject other DDL statements and resume the migration.

## Commands

When you use dmctl to manually handle the failed DDL statements, the commonly used commands include `query-status` and `binlog`.

### query-status

The `query-status` command is used to query the current status of items such as the subtask and the relay unit in each MySQL instance. For details, see [query status](/dm/dm-query-status.md).

### binlog

> The `binlog` command is used to manage and show binlog operations. This command is only supported in DM v6.0 and later versions. For earlier versions, use the `handle-error` command.

Reviewer 2: ```suggestion
The `binlog` command is used to manage and show binlog operations. This command is only supported in DM v6.0 and later versions. For earlier versions, the `handle-error` command can be used.
```
